### PR TITLE
[tests-only][full-ci] add tests to check versions of file in a space shared via link with viewer/editor role

### PR DESCRIPTION
**Pull request by prashant-gurung899**
## Description
This PR adds tests to check versions of a file inside a Personal and Project space that is shared via link with viewer and editor roles.
``` feature
Scenario Outline: check the versions of a file in a space shared via link as viewer/editor
Scenario Outline: check the versions of a file in a personal space shared via link as viewer
```
## Related Issue

- Related to: https://github.com/owncloud/ocis/issues/9955

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 1]
```diff
@@ -164,3 +164,33 @@ Feature: checking file versions using file id
       | source  | destination |
       | /       | folder/     |
       | folder/ | /           |
+
+
+  Scenario Outline: check the versions of a file in a space shared via link as viewer/editor
+    And user "Alice" has created the following space link share:
```
Comment (anon-pradip) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:170]:
```suggestion
    Given user "Alice" has created the following space link share:
```

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 2]
```diff
@@ -164,3 +164,33 @@ Feature: checking file versions using file id
...
+    When the public gets the number of versions of file "/text.txt" using password "%public%" using file-id path "/meta/<<FILEID>>/v"
+    Then the HTTP status code should be "403"
+    And the number of versions should be "0"
+    Examples:
+      | permissions-role |
+      | view             |
+      | edit             |
+
+
+  Scenario Outline: check the versions of a file in a personal space shared via link as viewer
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:183]:
```suggestion
  Scenario Outline: check the versions of a file in a personal space shared via link
```
if we include all possible role then maybe we can skip mentioning role name

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 3]
```diff
@@ -164,3 +164,33 @@ Feature: checking file versions using file id
       | source  | destination |
       | /       | folder/     |
       | folder/ | /           |
+
+
+  Scenario Outline: check the versions of a file in a space shared via link as viewer/editor
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:169]: same comment like below

**Review by amrita-shrestha — APPROVED** [review]
lgtm

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 4]
```diff
@@ -175,3 +175,34 @@ Feature: checking file versions using file id
...
     When user "Brian" gets the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
     Then the HTTP status code should be "403"
+
+
+  Scenario Outline: public tries to check the versions of a file in a project space shared via link as viewer/editor
+    Given user "Alice" has created the following space link share:
+      | space           | Project1           |
+      | permissionsRole | <permissions-role> |
+      | password        | %public%           |
+    When the public tries to get the number of versions of file "/text.txt" using password "%public%" using file-id path "/meta/<<FILEID>>/v"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:185]:
```suggestion
    When the public tries to get the number of versions of file "/text.txt" with password "%public%" using file-id path "/meta/<<FILEID>>/v"
```

File: tests/acceptance/bootstrap/FilesVersionsContext.php [hunk 5]
```diff
@@ -72,6 +72,32 @@ public function userGetsFileVersions(string $user, string $file):void {
...
+	/**
+	 * @When the public tries to get the number of versions of file :file using password :password using file-id path :endpoint
+	 *
+	 * @param string $file
+	 * @param string $password
+	 * @param string $endpoint
+	 *
+	 * @return void
+	 */
+	public function thePublicGetsTheNumberOfVersionsOfFileWithPasswordUsingFileIdPath(string $file, string $password, string $endpoint) {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/FilesVersionsContext.php:84]:
```suggestion
	public function thePublicGetsTheNumberOfVersionsOfFileWithPasswordUsingFileIdPath(string $file, string $password, string $endpoint): void {
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

**Review by saw-jan — APPROVED** [review]

**Pull request merged**